x-pack/plugins/apm/ftr_e2e/cypress/integration/power_user/feature_flag/infrastructure.spec.ts

```diff
@@ -31,17 +27,14 @@ describe.skip('Infrastracture feature flag', () => {
     await synthtrace.clean();
   });
 
-  beforeEach(() => {
-    cy.loginAsEditorUser();
-  });
-
   describe('when infrastracture feature is enabled', () => {
-    it('shows the flag as enabled in kibana advanced settings', () => {
-      cy.visit(settingsPath);
-
-      cy.get(infraToggle)
-        .should('have.attr', 'aria-checked')
-        .and('equal', 'true');
+    beforeEach(() => {
+      cy.loginAsEditorUser().then(() => {
+        // enables infrastructure view feature on advanced settings
+        cy.updateAdvancedSettings({
```

Comment:
How does cypress wait for these commands to finish? They are neither `return`'d or `await`'ed. Is it something magical in `cy`?